scripts/kind-setup.sh

```diff
@@ -0,0 +1,116 @@
+    _create_kind_cluster $__cluster_name $kubeconfig_path
+
+    info_msg "Exporting KUBECONFIG=$kubeconfig_path"
+    export KUBECONFIG=$kubeconfig_path
+
+    _install_additional_controllers $__controller_namespace
+}
+
+_create_kind_cluster() {
+    local __cluster_name=$1
+    local __kubeconfig_path=$2
+
+    local config_file_name="$(get_cluster_configuration_file_name)"
+    local cluster_version="$(get_cluster_k8s_version)"
+
+    local config_file_path=$SCRIPTS_DIR/kind-configurations/$config_file_name
+    
+    info_msg "Using configuration \"$config_file_name\""
+    debug_msg "Using K8s version \"$cluster_version\""
+
+    for i in $(seq 0 3); do
+        if [[ -z $(kind get clusters 2>/dev/null | grep $cluster_name) ]]; then
+            kind create cluster --name "$cluster_name" \
+                ${cluster_version:+ --image kindest/node:v$cluster_version} \
+                --config "$config_file_path" \
+                --kubeconfig $__kubeconfig_path 1>&2 || :
+        else
+            break
+        fi
+    done
+}
+
+_get_kind_cluster_name() {
+    local config_name=$(get_cluster_name)
+
+    if [[ "$config_name" == "" ]]; then
+        local name_uuid=$(uuidgen | cut -d'-' -f1 | tr '[:upper:]' '[:lower:]')
+        config_name="ack-test-${name_uuid}"
+    fi
+    echo $config_name
+}
+
+_install_additional_controllers() {
+    local __controller_namespace=$1
+
+    local install_region=$(get_aws_region)
+    local additional_controllers=( $(get_cluster_additional_controllers | yq -o=j -I=0 '.[]' -) )
+    for controller_version_pair in "${additional_controllers[@]}"; do
+        local controller_name=$(echo $controller_version_pair | tr -d '"' | cut -d "@" -f 1)
+        local controller_version=$(echo $controller_version_pair | tr -d '"' | cut -d "@" -f 2)
+
+        # Strip the `-controller` from the name
+        local controller_service=$(echo $controller_name | sed 's/-controller//')
+
+        if (_is_additional_controller_installed $controller_service $__controller_namespace); then
+            info_msg "$controller_name already installed. Skipping"
+        else
+            info_msg "Installing the $controller_name"
+            _install_additional_controller $controller_service $controller_version $install_region $__controller_namespace
+        fi
+    done
+}
+
+_is_additional_controller_installed() {
+    local __controller_service=$1
+    local __controller_namespace=$2
+    
+    local exists="$(helm list -q -n "$__controller_namespace" --filter "$__controller_service-chart+" 2>/dev/null)"
+    [[ "$exists" == "" ]] && return 1 || return 0
+}
+
+_install_additional_controller() {
+    local __controller_service=$1
+    local __controller_version=$2
+    local __region=$3
+    local __controller_namespace=$4
+
+    debug_msg "Logging into the Helm registry"
+    _perform_helm_login
+
+    helm install --create-namespace -n "$__controller_namespace" \
```

Comment:
Hmm actually I thought about this as I was pushing the code. We are installing the controllers but without any credentials, so they're essentially useless. We can't use IRSA because we can't assume they're running on EKS. However, we also can't use the `assumed_role_arn` because that might not have the IAM policies to run these controllers. Any ideas?

Reply:
I would say that we use the same assumed_role_arn, and permission of those role need to be extended when new cross-api references are added.

Another option would be managing multiple test iam roles for each AWS service (depending on which other controllers are installed), but i can see it getting messier to manage.